### PR TITLE
Better exceptions

### PR DESCRIPTION
**Pull request by AugustoPeres**
Starts on [this issue](https://github.com/inductiva/tasks/issues/59):

Prevents dumping very long stack to the user.

With:
```
Registering machine group configurations:
Error registering the machine group.
Failed with exception:
 (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': 'e4e6cd3a8faa07f6912d78dcdc5a598b', 'Date': 'Thu, 25 Jan 2024 17:17:20 GMT', 'Server': 'Google Frontend', 'Content-Length': '96'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 10. Requested: 80."}'
```

Without:

```
Registering machine group configurations:
Traceback (most recent call last):
  File "/Users/augustoperes/Documents/inductiva/../inductiva_test_scripts/launch_too_many_cores.py", line 3, in <module>
    mg = inductiva.resources.MachineGroup(machine_type="c2-standard-16",
  File "/Users/augustoperes/Documents/inductiva/inductiva/resources/machines.py", line 49, in __init__
    self._register_machine_group(num_vms=self.num_machines,
  File "/Users/augustoperes/Documents/inductiva/inductiva/resources/machines_base.py", line 83, in _register_machine_group
    resp = self._api.register_vm_group(body=instance_group_config)
  File "/Users/augustoperes/Documents/inductiva/inductiva/client/paths/compute_group/post.py", line 303, in register_vm_group
    return self._register_vm_group_oapg(
  File "/Users/augustoperes/Documents/inductiva/inductiva/client/paths/compute_group/post.py", line 219, in _register_vm_group_oapg
    raise exceptions.ApiException(status=response.status,
inductiva.client.exceptions.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': '9a0153fca058e663cf6da096de30ace5', 'Date': 'Thu, 25 Jan 2024 17:15:38 GMT', 'Server': 'Google Frontend', 'Content-Length': '96'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 10. Requested: 80."}'
```

### CODE REVIEW

File: inductiva/logs/#__init__.py#
Comment (sergiomsantos) [inline inductiva/logs/#__init__.py#:1]: what is this file?

File: inductiva/__init__.py [hunk 1]
```diff
@@ -18,11 +19,12 @@
...
 
 absl.logging.set_verbosity(absl.logging.INFO)
-logging.basicConfig(format="%(message)s")
 
 # Disable urllib3 warnings.
 # TODO: Verify and fix the appearance of this warning.
 urllib3_logger = logging.getLogger("urllib3.connectionpool")
 urllib3_logger.setLevel(logging.CRITICAL)
 
+logs.setup()
```
Comment (sergiomsantos) [inline inductiva/__init__.py:28]: move this to be the first thing that gets executed after the imports

**Comment by AugustoPeres** [comment]
### QA testing.

1. Install from this branch;
2. Launch a script that will fail, for example:
```python
import inductiva

mg = inductiva.resources.MachineGroup(machine_type="c2-standard-16",
                                      num_machines=2,
                                      disk_size_gb=70,
                                      register=True)
mg.start()

mg = inductiva.resources.MachineGroup(machine_type="c2-standard-16",
                                      num_machines=80,
                                      disk_size_gb=70,
                                      register=True)
mg.start()
```

The first machine group should be launched just fine.
The second should display the error:
```
Registering machine group configurations:
Registering machine group failed with exception (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': '5418a1ae0ac566c3b4a1c0fe291ea9f9;o=1', 'Date': 'Fri, 26 Jan 2024 10:55:21 GMT', 'Server': 'Google Frontend', 'Content-Length': '96'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 10. Requested: 80."}'

```

There should also exist a file `rotated.log`(in the directory from which you launched the script with):

```
2024-01-26 10:59:01|INFO|root|machines_base.py:82 Registering machine group configurations:
2024-01-26 10:59:03|INFO|root|machines_base.py:217 > Name: api-3rmf8k3o5ssjkbgzd6y89ewh3
2024-01-26 10:59:03|INFO|root|machines_base.py:218 > Machine Type: c2-standard-16
2024-01-26 10:59:03|INFO|root|machines_base.py:219 > Disk size: 70 GB
2024-01-26 10:59:03|INFO|absl|machines.py:78 > Number of machines: 2
2024-01-26 10:59:03|INFO|absl|machines.py:79 > Spot: False
2024-01-26 10:59:03|INFO|absl|machines.py:93 Estimated cloud cost for all machines : 1.838 $/h
2024-01-26 10:59:03|INFO|root|machines_base.py:136 Starting machine group. This may take a few minutes.
2024-01-26 10:59:03|INFO|root|machines_base.py:138 Note that stopping this local process will not interrupt the creation of the machine group. Please wait...
2024-01-26 10:59:27|INFO|root|machines_base.py:151 Machine group successfully started in 0:00:24.
2024-01-26 10:59:27|INFO|root|machines_base.py:82 Registering machine group configurations:
2024-01-26 10:59:28|ERROR|root|log.py:17 Registering machine group failed with exception (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'content-type': 'application/json', 'X-Cloud-Trace-Context': 'faf0b9b9189332476f6d65c5bf6524e1;o=1', 'Date': 'Fri, 26 Jan 2024 10:59:28 GMT', 'Server': 'Google Frontend', 'Content-Length': '96'})
HTTP response body: b'{"detail":"Maximum allowed number of machines is exceeded. Maximum allowed: 10. Requested: 80."}'
```

File: inductiva/logs/#__init__.py#
Comment (AugustoPeres) [reply inductiva/logs/#__init__.py#:1]: Ah, some weird emacs artifact that must have been logged by mistake

**Review by sergiomsantos — APPROVED** [review]

**Review by IvanPombo — APPROVED** [review]
It seems good to me! But just to understand better: is it automatic that it sends to the two different loggers now: stdout and file?

**Comment by AugustoPeres** [comment]
> It seems good to me! But just to understand better: is it automatic that it sends to the two different loggers now: stdout and file?

Yes

File: inductiva/resources/machines_base.py [hunk 2]
```diff
@@ -77,7 +80,13 @@ def _register_machine_group(self, **kwargs):
...
         )
         logging.info("Registering machine group configurations:")
-        resp = self._api.register_vm_group(body=instance_group_config)
+        try:
+            resp = self._api.register_vm_group(body=instance_group_config)
+        except (exceptions.ApiValueError, exceptions.ApiException) as e:
+            logs.log_and_exit(logging.getLogger(), logging.ERROR,
+                              "Registering machine group failed"\
+                              " with exception %s", e)
+        # resp = self._api.register_vm_group(body=instance_group_config)
```
Comment (IvanPombo) [inline inductiva/resources/machines_base.py:89]: Remove this.

**Comment by IvanPombo** [comment]
## QA

The purpose of the PR is handled correctly. When starting a machine with incorrect parameters, I get the following error:
Screenshot 2024-01-26 at 12 16 05

Then, for the registering I was able to get "Unhandled Exception" when inserting wrong value types:
Screenshot 2024-01-26 at 12 18 17

I guess this will need to be dealt in the Client, since I was able to register successfully a machine with `num_machines=-1`.

**Comment by sergiomsantos** [comment]
> I guess this will need to be dealt in the Client, since I was able to register successfully a machine with num_machines=-1.
@IvanPombo please open a bug

**Pull request merged**